### ISSUE TITLE
XSLT Saxon native support

### ISSUE THREAD

**Issue by ppalaga** [apache/camel-quarkus#1615]

**Comment by CynanX** [comment]
I hit this same issue when trying to build a native executable with `camel-quarkus-xslt-saxon`, though my use case is slightly different in that I exclude the home edition and import the personal edition.

Doing this I hit the same error as above. I then ran Quarkus along with the native agent to generate the required `reflect-config.json` file and can report that Saxon is now working fine for me in native mode.

**Comment by zhfeng** [comment]
@CynanX it would be very useful if you can share with ```reflect-ocnfig.json``` here. And we can take a look to improve in the build steps. Thanks!

**Comment by CynanX** [comment]
I'm currently using Quarkus 2.7.5.Final and building on GraalVM 21.3.1 Java 17 CE (Java Version 17.0.2+8-jvmci-21.3-b09).

The relevant sections of my pom are:

```
<dependency>
    <groupId>org.apache.camel.quarkus</groupId>
    <artifactId>camel-quarkus-xslt-saxon</artifactId>
    <exclusions>
        <exclusion>
            <groupId>net.sf.saxon</groupId>
            <artifactId>Saxon-HE</artifactId>
        </exclusion>
    </exclusions>
</dependency>

<dependency>
    <groupId>net.sf.saxon</groupId>
    <artifactId>saxon-pe</artifactId>
    <version>11.3</version>
</dependency>

<dependency>
    <groupId>com.sun.mail</groupId>
    <artifactId>javax.mail</artifactId>
    <version>1.6.2</version>
</dependency>
```

Here's is my `resource-config.json` file:

```
{
  "resources": {
    "includes": [
      {
        "pattern": "\\Qsaxon-license.lic\\E"
      }
    ]
  }
}
```

And finally here is my `reflect-config.json` file:

```
[
  {
    "name": "com.saxonica.config.ProfessionalConfiguration",
    "methods": [
      {
        "name": "<init>",
        "parameterTypes": []
      }
    ]
  },
  {
    "name": "com.saxonica.functions.extfn.Parse",
    "methods": [
      {
        "name": "<init>",
        "parameterTypes": []
      }
    ]
  },
  {
    "name": "net.sf.saxon.functions.StringJoin",
    "methods": [
      {
        "name": "<init>",
        "parameterTypes": []
      }
    ]
  },
  {
    "name": "net.sf.saxon.functions.String_1",
    "methods": [
      {
        "name": "<init>",
        "parameterTypes": []
      }
    ]
  },
  {
    "name": "net.sf.saxon.functions.Tokenize_1",
    "methods": [
      {
        "name": "<init>",
        "parameterTypes": []
      }
    ]
  },
  {
    "name": "org.apache.camel.component.xslt.XsltBuilder",
    "allDeclaredFields": true,
    "queryAllDeclaredMethods": true
  },
  {
    "name": "org.apache.camel.component.xslt.XsltComponent"
  },
  {
    "name": "org.apache.camel.component.xslt.XsltComponentConfigurer"
  },
  {
    "name": "org.apache.camel.component.xslt.XsltEndpointConfigurer"
  },
  {
    "name": "org.apache.camel.component.xslt.XsltEndpointUriFactory"
  },
  {
    "name": "org.apache.camel.component.xslt.saxon.XsltSaxonBuilder",
    "allDeclaredFields": true,
    "allDeclaredClasses": true,
    "queryAllDeclaredMethods": true,
    "methods": [
      {
        "name": "<init>",
        "parameterTypes": []
      }
    ]
  },
  {
    "name": "org.apache.camel.component.xslt.saxon.XsltSaxonComponent"
  },
  {
    "name": "org.apache.camel.component.xslt.saxon.XsltSaxonComponentConfigurer",
    "methods": [
      {
        "name": "<init>",
        "parameterTypes": []
      }
    ]
  },
  {
    "name": "org.apache.camel.component.xslt.saxon.XsltSaxonEndpointConfigurer",
    "methods": [
      {
        "name": "<init>",
        "parameterTypes": []
      }
    ]
  },
  {
    "name": "org.apache.camel.component.xslt.saxon.XsltSaxonEndpointUriFactory"
  },
  {
    "name": "org.xmlresolver.loaders.XmlLoader",
    "methods": [
      {
        "name": "<init>",
        "parameterTypes": [
          "org.xmlresolver.ResolverConfiguration"
        ]
      }
    ]
  },
  {
    "name": "org.xmlresolver.logging.DefaultLogger",
    "methods": [
      {
        "name": "<init>",
        "parameterTypes": [
          "org.xmlresolver.ResolverConfiguration"
        ]
      }
    ]
  }
]
```

Note. There are several more Saxon functions not in the file above but will be required it they are utilised by your XSLT files.

I saw a thread on StackOverFlow not long ago where by @michaelhkay mentioned they were looking at producing a native friendly version of Saxon, so he may be willing to assist here.

**Comment by michaelhkay** [comment]
Yes, we (more specifically my colleague O'Neil Delpratt) are actively working on getting Saxon to work under GraalVM; and as you've found, dynamic loading and reflection is the main challenge.

I'm not sure if we have anything more we can share with you at the moment. We're probably at about the same stage of development. But we're happy to share information if this helps us reach the goal more quickly.

Michael Kay
Saxonic

**Comment by ppalaga** [comment]
Thanks for the input, @CynanX. I am afraid, we won't be able to incorporate any Saxon PE and `com.saxonica.*` config because of licensing reasons. Anyway, to add a native support to an extension, the integration test is usually crucial. So if you have some camel-quarkus-xslt test to share a PR would be highly welcome.

**Comment by ppalaga** [comment]
Native support for XSLT Saxon was actually implemented via 66565dd0ea7ffd65184a3574fc20fd216fac5fdc so this issue might be seen as a duplicate of https://github.com/apache/camel-quarkus/issues/1710
